<p>
For better readability, do not put more than one statement on a single line.
</p>

<p>
The following code snippet illustrates this rule :
</p>

<h2>Noncompliant Code Example</h2>
<pre>
if(someCondition) doSomething(); // Noncompliant
</pre>


<h2>Compliant Solution</h2>
<pre>
if(someCondition)                // Compliant
  doSomething();
</pre>