<p>
  Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allow to check that all variable names match a provided regular expression.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass {
  public var FIRST: String; // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>
<pre>
public class MyClass {
  public var first: String; // Compliant
}
</pre>

