<p>
  Boolean expressions should not be compared against boolean literals, as their value can be directly used.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
  if (booleanVariable == true) { /* ... */ }
  if (booleanVariable != true) { /* ... */ }
</pre>

<h2>Compliant Solution</h2>
<pre>
  if (booleanVariable) { /* ... */ }
  if (!booleanVariable) { /* ... */ }
</pre>
