<p>
  Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
function doSomething(a:int, b:int):void      // "b" is unused
{
  compute(a);
}
</pre>

<h2>Compliant Solution</h2>
<pre>
function doSomething(a:int):void      // "b" is unused
{
  compute(a);
}
</pre>

<h2>Exceptions</h2>
<p>Methods overriding or implementing another one should be excluded</p>
<pre>
override function doSomething(a:int):void     // no issue reported on b
{
  compute(a);
}
</pre>
