<p>
  Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all package names match a provided regular expression.
</p>

<h2>Noncompliant Code Example</h2>
<p>The following code snippet illustrates this rule:</p>
<pre>
package org.Example { // Noncompliant
...
}
</pre>

<h2>Compliant Solution</h2>
<pre>
package org.example { // Compliant
...
}
</pre>
