<p>
  <code>for</code> loop stop conditions must be invariant (i.e. true at both the beginning and ending of every loop iteration). Ideally, this means that the stop condition is set to a local variable just before the loop begins.
</p>
<p>
  Stop conditions that are not invariant are difficult to understand and maintain, and will likely lead to the introduction of errors in the future.
</p>
<p>
  This rule tracks three types of non-invariant stop conditions:
</p>
<ul>
    <li>When the loop counters are updated in the body of the for loop</li>
    <li>When the stop condition depend upon a method call</li>
    <li>When the stop condition depends on an object property, since such properties could change during the execution of the loop.</li>
</ul>

<h1>Loop counter updated in the body of the <code>for</code> loop</h1>
<h2>Noncompliant Code Example</h2>
<p>Example of a loop counter updated in the body of the for loop:</p>
<pre>
for (var i = 0; i < 10; i++) {
  ...
  i = i - 1; // Noncompliant
  ...
}
</pre>


<h1>Stop condition depending upon the result of a method call</h1>
<h2>Noncompliant Code Example</h2>
<p>Example of a loop stop condition depending upon the result of a method call:</p>
<pre>
for (var i = 0; i < getMaximumNumber(); i++) {...}
</pre>

<h2>Compliant Solution</h2>
<pre>
int stopCondition = getMaximumNumber();
for (var i = 0; i < stopCondition; i++) {...}
</pre>

