p<p>
  Since ActionScript 3, the package definition should be done outside of the class definition even if the old syntax is still supported.
</p>

<h2>Noncompliant Code Example</h2>
<p>The following code snippet illustrates this rule with the default threshold of 3:</p>
<pre>
class P.A {   // Noncompliant
}
</pre>

<h2>Compliant Solution</h2>
<pre>
package P {
    class A {  // Compliant
  }
}
</pre>