<p>
  The <code>trace()</code> function outputs debug statements, which can be read by anyone with a debug version of the Flash player. Because sensitive information could easily be exposed in this manner, <code>trace()</code> should never appear in production code.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
var val:Number = doCalculation();
trace("Calculation result: " + val);  // Noncompliant
</pre>

<h2>Compliant Solution</h2>
<pre>
var val:Number = doCalculation();
</pre>
